File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/reactive/book/ReactiveBookEntity.java
package io.quarkus.it.mongodb.panache.reactive.book;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import javax.json.bind.annotation.JsonbDateFormat;

import org.bson.codecs.pojo.annotations.BsonIgnore;
import org.bson.codecs.pojo.annotations.BsonProperty;

import io.quarkus.it.mongodb.panache.book.BookDetail;
import io.quarkus.mongodb.panache.common.MongoEntity;
import io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoEntity;

@MongoEntity(collection = "TheBookEntity", clientName = "cl2")
public class ReactiveBookEntity extends ReactivePanacheMongoEntity {
    @BsonProperty("bookTitle")
    private String title;
    private String author;
    @BsonIgnore
    private String transientDescription;
    @JsonbDateFormat("yyyy-MM-dd")
    private LocalDate creationDate;

    private List<String> categories = new ArrayList<>();

    private BookDetail details;

    public String getTitle() {
        return title;
    }

    public ReactiveBookEntity setTitle(String title) {
        this.title = title;
        return this;
    }

    public String getAuthor() {
        return author;
    }

    public ReactiveBookEntity setAuthor(String author) {
        this.author = author;
        return this;
    }

    public List<String> getCategories() {
        return categories;
    }

    public ReactiveBookEntity setCategories(List<String> categories) {
        this.categories = categories;
        return this;
    }

    public BookDetail getDetails() {
        return details;
    }

    public ReactiveBookEntity setDetails(BookDetail details) {
        this.details = details;
        return this;
    }

    public String getTransientDescription() {
        return transientDescription;
    }

    public void setTransientDescription(String transientDescription) {
        this.transientDescription = transientDescription;
    }

    public LocalDate getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDate creationDate) {
        this.creationDate = creationDate;
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/reactive/book/ReactiveBookEntity.kt
package io.quarkus.it.mongodb.panache.reactive.book

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonFormat.Shape
import io.quarkus.it.mongodb.panache.book.BookDetail
import io.quarkus.mongodb.panache.common.MongoEntity
import io.quarkus.mongodb.panache.kotlin.reactive.ReactivePanacheMongoCompanion
import io.quarkus.mongodb.panache.kotlin.reactive.ReactivePanacheMongoEntity
import java.time.LocalDate
import java.util.ArrayList
import org.bson.codecs.pojo.annotations.BsonIgnore
import org.bson.codecs.pojo.annotations.BsonProperty

@MongoEntity(collection = "TheBookEntity", clientName = "cl2", readPreference = "primary")
class ReactiveBookEntity : ReactivePanacheMongoEntity() {
    companion object : ReactivePanacheMongoCompanion<ReactiveBookEntity>

    @BsonProperty("bookTitle")
    var title: String? = null
        private set

    var author: String? = null
        private set

    @BsonIgnore var transientDescription: String? = null

    @JsonFormat(shape = Shape.STRING, pattern = "yyyy-MM-dd") var creationDate: LocalDate? = null
    var categories: List<String> = ArrayList()
        private set

    var details: BookDetail? = null
        private set

    fun setTitle(title: String?): ReactiveBookEntity {
        this.title = title
        return this
    }

    fun setAuthor(author: String?): ReactiveBookEntity {
        this.author = author
        return this
    }

    fun setCategories(categories: List<String>): ReactiveBookEntity {
        this.categories = categories
        return this
    }

    fun setDetails(details: BookDetail?): ReactiveBookEntity {
        this.details = details
        return this
    }
}
